rust/auction/program/src/processor.rs

```diff
@@ -505,6 +582,18 @@ impl BidState {
                                 break;
                             }
                         }
+
+                        // Check if all the lots were sold with instant_sale_price
+                        if let Some(instant_sale_amount) = instant_sale_price {
+                            // bids.len() - max = index of the last winner bid
+                            if bids.len() >= *max
+                                && bids[bids.len() - *max].1 >= instant_sale_amount
+                            {
+                                msg!("All the lots were sold with instant_sale_price, auction is ended");
+                                *auction_state = AuctionState::Ended;
```

Comment:
Thinking about this, I think instead of passing in a mutable auction state, which is kind of a violation of interfaces and the law of demeter, we should instead have a function called place_bid on auction, that calls place bid on this, and it does this extra if statement check at that level, because that's the appropriate place for a change to auction status, not inside this other struct. Then you can do self.status = AuctionState::Ended. See the other wrapper methods in Auction already present. Also serves additional side effect of doing the check for all types of auctions as opposed to just english!